modules/sdk-core/src/bitgo/inscription/inscription.ts

```diff
@@ -0,0 +1,75 @@
+import assert from 'assert';
+import { IInscription, TxInfoUnspents } from './iInscription';
+import { BitGoBase } from '../bitgoBase';
+import { IWallet } from '../wallet';
+import { inscriptions, PreparedInscriptionRevealData, TapLeafScript } from '@bitgo/utxo-ord';
+import { KeyIndices } from '../keychain';
+import { xprvToRawPrv } from '../../account-lib';
+import { HalfSignedUtxoTransaction } from '../baseCoin';
+import { inferNetworkFromCoinName } from '../utils';
+
+export class Inscription implements IInscription {
+  private readonly bitgo: BitGoBase;
+  private readonly wallet: IWallet;
+
+  constructor(bitgo: BitGoBase, wallet: IWallet) {
```

Comment:
let's pass in `coin: AbstractUtxoCoin` here so we can use `this.coin.network` normally

Reply:
`this.coin = getUtxoCoin(wallet.coin());`

This is typed as an `AbstractUtxoCoin`

Reply:
AbstractUtxoCoin type is not imported in sdk-core, I can't do this.

Reply:
`inscription.ts` should live in `AbstractUtxoCoin`. It is a utxo-specific concept after all.

Reply:
Ideally yes, but these methods need to be called within a wallet for our users. We run into this same issue where we only have basecoin to work with in sdk-core. We still won't be able to do something like wallet.coin().inscriptionMethodHere().

Reply:
ok can't we have a bunch of methods in `AbstractUtxoCoin` then that take the wallet as a parameter?

Reply:
Added it to abstract utxo coin. baseCoin still needs this since we need to keep in mind UI usage, we do not use coin specific methods.